tests/cephadm/test_cluster_view.py

```diff
@@ -0,0 +1,88 @@
+from json import loads
+from time import sleep
+
+from cli.cephadm.cephadm import CephAdm
+from utility.log import Log
+
+log = Log(__name__)
+
+
+class ClusterViewError(Exception):
+    pass
+
+
+def verify_mon_ips(host):
+    """
+    Verify MON IPs updated in ceph.conf
+    Args:
+        host: Installer host
+    """
+    # Get updated MON nodes
+    cmd = "cephadm shell ceph orch ps --daemon-type mon --format json"
+    out = host.exec_command(sudo=True, cmd=cmd)[0]
+    configs = loads(out)
+    mon_nodes = [config.get("daemon_id") for config in configs]
+
+    # Get MON node IPs
+    mon_ips = []
+    out = CephAdm(host).ceph.orch.host.ls(format="json")
+    configs = loads(out)
+    for config in configs:
+        if config.get("hostname") in mon_nodes:
+            mon_ips.append(config.get("addr"))
+
+    # Validate MON IPs updated in ceph.conf
+    mon_ips_conf = host.remote_file(file_name="/etc/ceph/ceph.conf", file_mode="r")
+    mon_ips_conf = mon_ips_conf.readlines()[-1]
+    for mon_ip in mon_ips:
+        if mon_ip not in mon_ips_conf:
+            raise ClusterViewError("MON IPs not updated in ceph.conf")
+
+
+def run(ceph_cluster):
+    """Verify admin node has updated view of cluster
+    Args:
+        ceph_cluster: CephNode or list of CephNode object
+    """
+
+    install_node = ceph_cluster.get_nodes(role="installer")[0]
+
+    # Apply MON placement
+    conf = {"pos_args": [], "placement": "5"}
+    out = CephAdm(install_node).ceph.orch.apply(service_name="mon", **conf)
+    if "Scheduled mon update" not in out:
+        raise ClusterViewError("Apply MON placement Failed")
+    sleep(60)
+
+    # Check Ceph health status
+    out = CephAdm(install_node).ceph.status()
+
+    # ls -lrt /etc/ceph/
+    dir_out = install_node.get_dir_list(dir_path="/etc/ceph/")
+    if "ceph.conf" not in dir_out:
+        raise ClusterViewError("Ceph.conf not found in /etc/ceph/")
+
+    # Verify MON IPs update in ceph.conf
+    verify_mon_ips(install_node)
+
+    # Get non-MON nodes
+    non_mon_nodes = []
+    nodes = ceph_cluster.get_nodes()
+    for node in nodes:
+        if node.role not in ["mon", "client"]:
+            non_mon_nodes.append(node)
+
+    # Apply MON on non OSD nodes
+    conf = {
+        "pos_args": [],
+        "host": f"\"{','.join([str(node) for node in non_mon_nodes])}\"",
+    }
+    out = CephAdm(install_node).ceph.orch.apply(service_name="mon", **conf)
+    if "Scheduled mon update" not in out:
+        raise ClusterViewError("Apply mon placement Failed")
+    sleep(60)
```

Comment:
Please avoid fixed timeouts, you can use WaitUntil, for e.g: Refer : https://github.com/red-hat-storage/cephci/blob/40201c55f9a165a7af12811d268294a00af20d38/tests/cephadm/test_remove_service.py#L24

Reply:
Thanks, I'll use the WaitUntil!

Reply:
@pranavprakash20 I've updated the changes please have a look.